index.js

```diff
@@ -35,16 +35,20 @@ var rethinkdbInit = function (r) {
    */
   var mapIndexes = function (table, db, conn) {
     return table.indexes.map(function (index) {
-     if (typeof index !== 'object' && typeof index !== 'string') throw new TypeError('index entry in table entry must be `Object` or `String`');
-     if (typeof index === 'string') return r.db(db).table(table.name).indexCreate(index).run(conn).catch(existsHandler);
-     if (index.name === undefined) throw new TypeError('index entry object in table schema must have a `name` property');
-     var opts = [];
-     if (index.indexFunction) opts.push(index.indexFunction);
-     if (index.multi || index.geo) opts.push(_.pick(index, ['multi', 'geo']));
-     return r.db(db).table(table.name)
-       .indexCreate(index.name, opts[0], opts[1])
-       .run(conn)
-       .catch(existsHandler);
+      if (typeof index !== 'object' && typeof index !== 'string') throw new TypeError('index entry in table entry must be `Object` or `String`');
+      if (typeof index === 'string') return r.db(db).table(table.name).indexCreate(index).run(conn).catch(existsHandler);
+      if (index.name === undefined) throw new TypeError('index entry object in table schema must have a `name` property');
+      var opts = [];
+      if (index.indexFunction) opts.push(index.indexFunction);
+      if (index.multi || index.geo) opts.push(_.pick(index, ['multi', 'geo']));
+
+      return r.db(db).table(table.name)
+        .indexCreate(index.name, opts[0], opts[1])
+        .run(conn)
+        .catch(existsHandler)
+        .then(function () {
+          return r.db(db).table(table.name).indexWait().run(conn);
```

Comment:
@thejsj I added this line because without it some tests would occasionally fail, for example:

```
ReqlOpFailedError: Index `vals` on table `rethinkdb_init_test.table_2` was accessed before its construction was finished in:
r.db("rethinkdb_init_test").table("table_2").getAll(1, {"index": "vals"}).coerceTo("array")
```

Depending on how people are using `rethinkdb-init`, this may be undesirable so let me know what you think and if we should just do the `indexWait` in the tests instead.